src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -56,19 +79,58 @@ public Command parseCommand(String input) {
             return new IncorrectCommand(MESSAGE_INVALID_COMMAND_FORMAT);
         }
         String commandWord = matcher.group(COMMAND_WORD_GROUP).toLowerCase().trim();
+        String commandFlag = matcher.group(COMMAND_FLAG_GROUP).toLowerCase();
         String parameters = matcher.group(PARAMETERS_GROUP);
 
         try {
             switch (commandWord){
-            case EditModuleCommand.COMMAND_WORD:
-                return prepareEditModuleCommand(parameters);
+            case COMMAND_WORD_EDIT:
+                return getEditCommand(commandFlag, parameters);
+            case COMMAND_WORD_ADD:
+                return getAddCommand(commandFlag, parameters);
+            case COMMAND_WORD_DELETE:
+                return getDeleteCommand(commandFlag, parameters);
+            case COMMAND_WORD_DONE:
+                return DoneCommand(parameters); //parameters is the index
+            case COMMAND_WORD_LIST:
+                return ListCommand(commandFlag); //command flag is the -t or -m
             default:
                 return null;
             }
         } catch (InvalidParameterException e) {
             return new IncorrectCommand(MESSAGE_INVALID_PARAMETERS);
         }
+        return null;
+    }
+
+    private DeleteCommand getDeleteCommand(String commandFlag, String parameters) {
+        if (commandFlag.equals(MODULE_PREFIX)) {
+            return new DeleteCommand(typeOfEntries.MODULE, null , parameters); //parameter is module code
```

Comment:
Need to remove the null due to updated constructor for DeleteCommand. 1 more occurence below.